spec/models/user_spec.rb

```diff
@@ -127,6 +127,54 @@
     end
   end
 
+  describe "#update" do
+    let(:existing_user) { create(:user) }
+
+    it "sets renamed_at if username is changed" do
+      freeze_time
+      existing_user.update(login: "new_username")
+      expect(existing_user.renamed_at).to eq(Time.current)
+    end
+
+    it "does not set renamed_at when username is not changed" do
+      existing_user.update(email: "newemail@example.com")
+      expect(existing_user.renamed_at).to be_nil
+    end
+
+    context "username recently changed" do
+      before do
+        freeze_time
+        existing_user.update(login: "new_login")
+      end
+
+      it "does not allow another rename" do
+        expect(existing_user.update(login: "new")).to be_falsey
+        localized_renamed_at = I18n.l(Time.now.in_time_zone(existing_user.renamed_at.time_zone), format: :long)
```

Comment:
Since we froze time before the test, instead of `Time.now.in_time_zone(existing_user.renamed_at.time_zone)`, does `existing_user.renamed_at` work?